subsys/bluetooth/host/sdp.c

```diff
@@ -2217,6 +2217,9 @@ static int bt_sdp_get_attr(const struct net_buf *buf,
 		}
 
 		data += sizeof(uint8_t);
+		if ((data + sizeof(uint16_t) - buf->data) > buf->len) {
+			return -EINVAL;
+		}
```

Comment:
Ideally this whole function should be refactored into using the buffer API to extract data and increment the data pointer, but that's outside the scope of this PR. 

However, I'll suggest to use `sizeof(id)` to better describe why we checking for a 16-bit length. 
```suggestion
		if ((data + sizeof(id) - buf->data) > buf->len) {
			return -EINVAL;
		}
```

Reply:
Updated, Thanks.